File: androidApp/src/main/java/com/george/kmpmoviepoc/android/PosterUrl.kt
package com.george.kmpmoviepoc.android

fun posterUrl(imdbId: String, height: Int): String? {
    // O(1): builds a URL string.
    val apiKey = BuildConfig.OMDB_POSTER_API_KEY
    if (apiKey.isBlank()) return null
    return "https://img.omdbapi.com/?i=$imdbId&h=$height&apikey=$apiKey"
}

File: iosApp/iosApp/PosterUrl.swift
import Foundation

func posterUrl(imdbId: String, height: Int) -> URL? {
    // O(1): builds a URL string.
    let apiKey = AppConfig.omdbPosterApiKey
    if apiKey.isEmpty || apiKey == "YOUR_OMDB_API_KEY" {
        return nil
    }
    let urlString = "https://img.omdbapi.com/?i=\(imdbId)&h=\(height)&apikey=\(apiKey)"
    return URL(string: urlString)
}
